src/Controller/Component/MaintenanceComponent.php

```diff
@@ -99,15 +102,15 @@ public function beforeFilter()
                 header('HTTP/1.0 401 Unauthorized');
                 echo 'Unauthorized';
                 exit;
-            } else {
-                if ($_SERVER['PHP_AUTH_USER'] == $user && $_SERVER['PHP_AUTH_PW'] == $password) {
-                    $this->_controller->Cookie->write($cookieName, true);
+            }
 
-                    return $this->_controller->redirect($successUrl);
-                }
+            if ($_SERVER['PHP_AUTH_USER'] == $user && $_SERVER['PHP_AUTH_PW'] == $password) {
```

Comment:
There should be APIs from Cake to achive that.

Reply:
`BasicAuthenticate` uses these headers, but is out of the scope of this Pull Request.

But let's try this new feature of github (added type safety):

```suggestion
            if ($_SERVER['PHP_AUTH_USER'] === $user && $_SERVER['PHP_AUTH_PW'] === $password) {
```